src/seraphis_main/enote_finding_context.cpp

```diff
@@ -0,0 +1,191 @@
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+//paired header
+#include "enote_finding_context.h"
+
+//local headers
+#include "async/misc_utils.h"
+#include "device/device.hpp"
+#include "seraphis_main/contextual_enote_record_types.h"
+#include "seraphis_main/scan_balance_recovery_utils.h"
+
+//third party headers
+
+//standard headers
+
+
+#undef MONERO_DEFAULT_LOG_CATEGORY
+#define MONERO_DEFAULT_LOG_CATEGORY "seraphis"
+
+namespace sp
+{
+//-------------------------------------------------------------------------------------------------------------------
+void EnoteFindingContextLegacySimple::view_scan_chunk(const LegacyUnscannedChunk &legacy_unscanned_chunk,
+    sp::scanning::ChunkData &chunk_data_out)
+{
+    for (const auto &blk : legacy_unscanned_chunk)
+    {
+        for (const auto &tx : blk.unscanned_txs)
+        {
+            // Identify owned enotes
+            if (tx.enotes.size() > 0)
+            {
+                std::list<sp::ContextualBasicRecordVariant> collected_records;
+                sp::scanning::try_find_legacy_enotes_in_tx(m_legacy_base_spend_pubkey,
+                    m_legacy_subaddress_map,
+                    m_legacy_view_privkey,
+                    blk.block_index,
+                    blk.block_timestamp,
+                    tx.transaction_id,
+                    tx.total_enotes_before_tx,
+                    tx.unlock_time,
+                    tx.tx_memo,
+                    tx.enotes,
+                    sp::SpEnoteOriginStatus::ONCHAIN,
+                    hw::get_device("default"),
+                    collected_records);
+
+                chunk_data_out.basic_records_per_tx[tx.transaction_id] = std::move(collected_records);
+            }
+            else
+            {
+                // always add an entry for tx in the legacy basic records map (since we save key images for every tx)
+                chunk_data_out.basic_records_per_tx[tx.transaction_id] = std::list<sp::ContextualBasicRecordVariant>{};
+            }
+
+            // Collect key images
+            sp::SpContextualKeyImageSetV1 collected_key_images;
+            if (sp::scanning::try_collect_key_images_from_tx(blk.block_index,
+                    blk.block_timestamp,
+                    tx.transaction_id,
+                    tx.legacy_key_images,
+                    std::vector<crypto::key_image>()/*sp_key_images*/,
+                    sp::SpEnoteSpentStatus::SPENT_ONCHAIN,
+                    collected_key_images))
+            {
+                chunk_data_out.contextual_key_images.emplace_back(std::move(collected_key_images));
+            }
+        }
+    }
+}
+//-------------------------------------------------------------------------------------------------------------------
+void EnoteFindingContextLegacyMultithreaded::view_scan_chunk(const LegacyUnscannedChunk &legacy_unscanned_chunk,
```

Comment:
On my machine with 8 threads and a pending chunk queue of size 10, there is no benefit to using this multithreaded enote finding context in the async scanner. My CPU is already maxxed out.

This enote finding context should only be useful when the pending chunk queue size is < number of threads available on the machine. Example scenario: when pointing the scanner to a daemon not running the changes from this PR, it's more efficient to use a pending chunk queue size of 1, and to use this multithreaded enote finding context, since the daemon does not yet know about the `max_block_count` request param. The perf gain for client-side scanning in this case is then sped up by a multiple of the number of threads on the machine.